moto/ec2/models.py

```diff
@@ -1444,6 +1457,13 @@ def attach_volume(self, volume_id, instance_id, device_path):
             return False
 
         volume.attachment = VolumeAttachment(volume, instance, device_path)
+        # Modify instance to capture mount of block device.
+        bdt = BlockDeviceType(volume_id=volume_id, status=volume.status, size=volume.size,
+                              attach_time=utc_date_and_time())
+        try:
+            instance.block_device_mapping[device_path] = bdt
+        except:
```

Comment:
The setdefault doesn't clear out the /dev/sda1 key which is introduced on a launch instance in moto (i call it a ghost volume - as it doesn't exist when describe_all_volumes() but shows up when looking at block_device_mapping of the instance - initially was going to deleted but decided to keep for backward compatability). So I just used a try/except to get around it, but could have dealt with it specifically for /dev/sda1 I guess.